A11y fixes

SamPotasz opened the pull request:
Fixes issues #1 #2 #3 #4 and #5

SamPotasz committed:
changed result highlighted word to link

SamPotasz committed:
Changed highlighted word to button for improved accessibility and keyboard focus

SamPotasz committed:
cleaned up main section on index.html to fix a11y errors

SamPotasz committed:
Removed placeholder text per USWDS guidelines

SamPotasz committed:
- Hiding alert box before clicking on ableist word
- scrolling to alert box after clicking word and alert box is off screen

SamPotasz committed:
- fixed spacing issues on About page
- Added default styling to <p> tags and removed inline styling
- Fixed markup in alert list on results page
- Fixed input text area starting with lots of spaces input

SamPotasz committed:
- added skipnav to header
- moved all of nav into layout template instead of separating into each page
- defined currPage variable in app.py to highlight current page in navbar

SamPotasz committed:
fixed styling issues on govbanner and navbar

SamPotasz committed:
removed erroneous <main> and #main-content elements from pages outside of layout.html

SamPotasz commented:
> Fixes issues #1 #2 #3 #4 and #5

Also now fixes #7 and #8

dianalam reviewed (commented):
thanks! the buttons look good! I noticed that the fixes changed the layout of the homepage (the header and main are now misaligned, "get started" became "find ableist language in your job posting", and the "how to use this tool" moved out of the blue box and into the main body text). is there a reason you made these changes?

![image](https://user-images.githubusercontent.com/13712704/132027194-0c3af95a-1e29-4275-8f8a-58c58e301294.png)

SamPotasz committed:
Aligned main to header on desktop and wider

SamPotasz committed:
Footer now aligns with rest of page

SamPotasz commented:
Ah, my bad on the navbar misalignment. Should be fixed with the latest push.

As for the headers, there wasn't an `<h1>` on the page before, and best practice is to have (exactly) one on each page. I did my best for the homepage but please edit it as you see fit!

I moved the explanatory text out of the alert box because to me it made the flow of the document hard to follow, and the way it was laid out it sometimes appeared above the text and sometimes below. Seemed to make sense to just put it as intro text above the form at all times.

Upon typing this out, maybe it makes sense to just have "Getting Started" be the h1, eliminate other subheadings, and keep the explanatory text above the form.

How does that seem?

Commit:
Consolidate headers in home page

dianalam commented:
Thanks! the `h1` recommendation makes sense. I just pushed some edits to try and consolidate the headers: @SamPotasz and @mtpatter: lmk what you think. @SamPotasz - would it be possible to adjust the line spacing under the headers? Seems like there's now too much space under "how to use this tool" and "get started." The font size of "paste the job description..." also seems bigger than the font size of the explanatory paragraph above--could we fix this? 

And one other thing I noticed - the text box on the home page is now much narrower than the one on the results page; can we increase the size of the text box on the home page so that they're consistent?

SamPotasz committed:
- Fixed text size of input label
- Fixed spacing around headers on index.html
- Fixed width of text area to match width of results paragraph

SamPotasz commented:
Thanks, @dianalam :-) I just pushed a few changes to fix the spacing, text size, and textarea width issues you mentioned.

I think the formatting should be good to go now. I modified the text of the H1 because the paragraph right below didn't seem to answer the question "How to use this tool," but I'll leave any further content changes up to the team!

dianalam commented:
@SamPotasz Thanks! Changes look great. 

I noticed one more thing: On the results page, the yellow text box with suggestions now gets appended to after every click instead of being replaced with the information for the word that was clicked on. Did one of the a11y/formatting changes generate this effect?

![image](https://user-images.githubusercontent.com/13712704/133142298-c5773f84-4586-4c80-8233-c14ca182761f.png)

SamPotasz committed:
Fixed error where multiple suggestions would extend the info box on /predict
- Moved javascript to separate file

SamPotasz committed:
Merge branch 'a11y-fixes' of https://github.com/USDepartmentofLabor/ableist-language-detector-web into a11y-fixes

SamPotasz commented on static/main.js:
```diff
@@ -0,0 +1,27 @@
+    toAdd += `<li class="margin-bottom-1">${verb}</li>`
+  }
+  toAdd += ul_end + example
+  outputEl.innerHTML = toAdd
```
previous line was `+= toAdd` which caused the error

dianalam commented:
thanks!
